File: problemset-new/008/00852-peak-index-in-a-mountain-array/freedomDR.go
package main

import "fmt"

func peakIndexInMountainArray(A []int) int {
	lo, hi := 0, len(A)
	for lo < hi {
		mid := lo + (hi-lo)/2
		if A[mid] < A[mid+1] {
			lo = mid + 1
		} else {
			hi = mid
		}
	}
	return lo
}
func main() {
	fmt.Println("vim-go")
}

File: problemset-new/008/00852-peak-index-in-a-mountain-array/freedomDR.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& A) {
        int lo = 0, hi = A.size();
        while(lo < hi) {
            int mid = lo + (hi-lo)/2;
            if(A[mid] < A[mid+1]) lo = mid+1;
            else hi = mid;
        }
        return lo;
    }
};
int main()
{
    return 0;
}
